src/msw/graphicsd2d.cpp

```diff
@@ -360,6 +360,221 @@ HRESULT WINAPI wxD3D11CreateDevice(
 }
 #endif
 
+#if wxUSE_PRIVATE_FONTS
+
+// This function is defined in src/msw/font.cpp.
+extern const wxArrayString& wxGetPrivateFontFileNames();
+
+namespace
+{
+    IDWriteFontCollection* gs_pPrivateFontCollection = NULL;
+
+    typedef unsigned int wxDirect2DFontKey;
+
+    class wxDirect2DFontFileEnumerator : public IDWriteFontFileEnumerator
+    {
+    public:
+        wxDirect2DFontFileEnumerator(IDWriteFactory* pFactory)
+            : m_refCount(0)
+            , m_factory(pFactory)
+            , m_nextIndex(0)
+        {
+        }
+
+        HRESULT Initialize(wxArrayString& collection)
+        {
+            if ( collection.empty() )
+                return E_INVALIDARG;
+
+            m_filePaths = collection;
+            return S_OK;
+        }
+
+        ~wxDirect2DFontFileEnumerator()
+        {
+        }
+
+        // IUnknown methods
+        virtual HRESULT STDMETHODCALLTYPE QueryInterface(REFIID iid, void** ppvObject) wxOVERRIDE
+        {
+            if ( IsEqualIID(iid, IID_IUnknown) || IsEqualIID(iid, __uuidof(IDWriteFontFileEnumerator)) )
+            {
+                *ppvObject = this;
+                AddRef();
+                return S_OK;
+            }
+            else
+            {
+                *ppvObject = NULL;
+                return E_NOINTERFACE;
+            }
+        }
+
+        virtual ULONG STDMETHODCALLTYPE AddRef() wxOVERRIDE
+        {
+            return ::InterlockedIncrement(&m_refCount);
+        }
+
+        virtual ULONG STDMETHODCALLTYPE Release() wxOVERRIDE
+        {
+            ULONG newCount = ::InterlockedDecrement(&m_refCount);
+            if ( newCount == 0 )
+                delete this;
+
+            return newCount;
+        }
+
+        // IDWriteFontFileEnumerator methods
+        virtual HRESULT STDMETHODCALLTYPE MoveNext(BOOL* pHasCurrentFile) wxOVERRIDE
+        {
+            HRESULT hr = S_OK;
+
+            *pHasCurrentFile = FALSE;
+            m_currentFile.reset();
+
+            if ( m_nextIndex < m_filePaths.size() )
+            {
+                hr = m_factory->CreateFontFileReference(m_filePaths[m_nextIndex].wc_str(), NULL, &m_currentFile);
+                if ( SUCCEEDED(hr) )
+                {
+                    *pHasCurrentFile = TRUE;
+                    ++m_nextIndex;
+                }
+            }
+
+            return hr;
+        }
+
+        virtual HRESULT STDMETHODCALLTYPE GetCurrentFontFile(IDWriteFontFile** ppFontFile) wxOVERRIDE
+        {
+            if ( m_currentFile )
+            {
+                m_currentFile.get()->AddRef();
+            }
+            *ppFontFile = m_currentFile;
+
+            return (m_currentFile != NULL) ? S_OK : E_FAIL;
```

Comment:
Extremely minor, but:
```suggestion
            return m_currentFile ? S_OK : E_FAIL;
```
`!= NULL` is inconsistent with omitting it just above and the parentheses are just unnecessary.